plugins/modules/crypto_info.py

```diff
@@ -0,0 +1,335 @@
+    if has_x25519:
+        try:
+            from cryptography.hazmat.primitives.asymmetric.x25519 import X25519PrivateKey
+            if CRYPTOGRAPHY_HAS_X25519_FULL:
+                X25519PrivateKey.from_private_bytes(b'')
+            else:
+                # Some versions do not support serialization and deserialization - use generate() instead
+                X25519PrivateKey.generate()
+        except ValueError:
+            pass
+        except UnsupportedAlgorithm:
+            has_x25519 = False
+
+    has_x448 = CRYPTOGRAPHY_HAS_X448
+    if has_x448:
+        try:
+            from cryptography.hazmat.primitives.asymmetric.x448 import X448PrivateKey
+            X448PrivateKey.from_private_bytes(b'')
+        except ValueError:
+            pass
+        except UnsupportedAlgorithm:
+            has_x448 = False
+
+    curves = []
+    if CRYPTOGRAPHY_HAS_EC:
+        import cryptography.hazmat.backends
+        import cryptography.hazmat.primitives.asymmetric.ec
+
+        backend = cryptography.hazmat.backends.default_backend()
+        for curve_name, constructor_name in CURVES:
+            ecclass = cryptography.hazmat.primitives.asymmetric.ec.__dict__.get(constructor_name)
+            if ecclass:
+                try:
+                    cryptography.hazmat.primitives.asymmetric.ec.generate_private_key(curve=ecclass(), backend=backend)
+                    curves.append(curve_name)
+                except UnsupportedAlgorithm:
+                    pass
+
+    info = {
+        'version': CRYPTOGRAPHY_VERSION,
+        'curves': curves,
+        'has_ec': CRYPTOGRAPHY_HAS_EC,
+        'has_ec_sign': CRYPTOGRAPHY_HAS_EC_SIGN,
+        'has_ed25519': has_ed25519,
+        'has_ed25519_sign': has_ed25519 and CRYPTOGRAPHY_HAS_ED25519_SIGN,
+        'has_ed448': has_ed448,
+        'has_ed448_sign': has_ed448 and CRYPTOGRAPHY_HAS_ED448_SIGN,
+        'has_dsa': CRYPTOGRAPHY_HAS_DSA,
+        'has_dsa_sign': CRYPTOGRAPHY_HAS_DSA_SIGN,
+        'has_rsa': CRYPTOGRAPHY_HAS_RSA,
+        'has_rsa_sign': CRYPTOGRAPHY_HAS_RSA_SIGN,
+        'has_x25519': has_x25519,
+        'has_x25519_serialization': has_x25519 and CRYPTOGRAPHY_HAS_X25519_FULL,
+        'has_x448': has_x448,
+    }
+    result['python_cryptography_capabilities'] = info
+    return result
+
+
+def add_openssl_information(module):
+    openssl_binary = module.get_bin_path('openssl')
+    result = {
+        'openssl_present': openssl_binary is not None,
+    }
+    if openssl_binary is None:
+        return result
+
+    openssl_result = {
+        'path': openssl_binary,
+    }
+    result['openssl'] = openssl_result
+
+    rc, out, err = module.run_command([openssl_binary, 'version'])
+    if rc == 0:
+        openssl_result['version_output'] = out
+        parts = out.split(None, 2)
+        if len(parts) > 1:
+            openssl_result['version'] = parts[1]
+
+    return result
+
+
+INFO_FUNCTIONS = [
+    add_crypto_information,
+    add_openssl_information,
+]
```

Comment:
```suggestion
INFO_FUNCTIONS = (
    add_crypto_information,
    add_openssl_information,
)
```
Same for `CURVES`.
Not important, but makes me feel better about using globals :)